src/v/cloud_topics/level_zero/reader/fetch_request_handler.cc

```diff
@@ -125,7 +125,24 @@ ss::future<> fetch_handler::process_single_request(l0::read_request<>* req) {
             co_return;
         }
 
-        auto data = std::move(extent.get());
+        auto res = extent.get();
+        if (res.has_error()) {
+            vlog(
+              req->rtc_logger.warn,
+              "Failed to materialize placeholders, error: {} ({})",
+              res.error(),
+              res.error().message());
+            std::error_code ec = res.error();
+            if (ec.category() == error_category()) {
```

Comment:
nit: it'd be nice if `error_category` could be something more specific like maybe even just repeating the scope like `cloud_topics::error_category`.